server/db/models/index.js

```diff
@@ -3,12 +3,17 @@ const User = require('./user')
 const Category = require('./category')
 const Product = require('./product')
 const Reviews = require('./reviews')
+const Cart = require('./cart')
 
 const ProductCategory = db.define('productCategory')
-
 Category.belongsToMany(Product, {through: ProductCategory})
 Product.belongsToMany(Category, {through: ProductCategory})
 
+const CartProduct = db.define('cartProduct')
+Product.belongsToMany(Cart, {through: CartProduct})
+Cart.belongsToMany(Product, {through: CartProduct})
+Cart.belongsTo(User, {constraints: false, foreignKeyConstraint: false}) //allow a null foreignkey userId to exist on an instance of Cart
```

Comment:
I think we should also add the product quantity to the join table to persist this info in the cart